source/session.ts

```diff
@@ -705,7 +707,7 @@ export class Session {
           entityType,
           primaryKeys.map((key: string) => updateEntity[key]),
           Object.keys(data).reduce<T>((accumulator, key: keyof T) => {
-            if (primaryKeys.indexOf(key) === -1) {
+            if (primaryKeys.indexOf(key.toString()) === -1) {
```

Comment:
is this really necessary? can't we do something like keyof T extends String?

Reply:
We can type cast key as a string, but then I think this is more proper